pex/cli/commands/lock.py

```diff
@@ -220,3 +277,149 @@ def _export(self):
+                assume_manylinux=target_configuration.assume_manylinux,
+            )
+            result = catch(
+                lock_updater.update_resolve,
+                locked_resolve=locked_resolve,
+                target_configuration=target_configuration,
+            )
+            if isinstance(result, Error):
+                error_by_target[target] = result
+            else:
+                platform = target.get_supported_tags()[0]
+                lock_updates_by_platform[platform] = result.updates
+                locked_resolve_by_platform[platform] = result.updated_resolve
+
+        if error_by_target:
+            return Error(
+                "Encountered {count} {errors} updating {lockfile_path}:\n{error_details}".format(
+                    count=len(error_by_target),
+                    errors=pluralize(error_by_target, "error"),
+                    lockfile_path=lockfile_path,
+                    error_details="\n".join(
+                        "{index}.) {platform}: {error}".format(
+                            index=index, platform=target.get_supported_tags()[0], error=error
+                        )
+                        for index, (target, error) in enumerate(error_by_target.items(), start=1)
+                    ),
+                ),
+            )
+
+        output = sys.stdout if dry_run else sys.stderr
+        for platform, lock_updates in lock_updates_by_platform.items():
+            for project_name, version_update in lock_updates.items():
+                if version_update:
+                    print(
+                        "{lead_in} {project_name} from {original_version} to {updated_version} in "
+                        "lock generated by {platform}.".format(
+                            lead_in="Would update" if dry_run else "Updated",
+                            project_name=project_name,
+                            original_version=version_update.original,
+                            updated_version=version_update.updated,
+                            platform=platform,
+                        ),
+                        file=output,
+                    )
+                else:
+                    print(
+                        "There {tense} no updates for {project_name} in lock generated by "
+                        "{platform}.".format(
+                            tense="would be" if dry_run else "were",
+                            project_name=project_name,
+                            platform=platform,
+                        ),
+                        file=output,
+                    )
+        if not dry_run:
+            lockfile.store(
+                attr.evolve(
+                    lock_file,
+                    pex_version=__version__,
+                    locked_resolves=SortedTuple(locked_resolve_by_platform.values()),
+                ),
+                lockfile_path,
+            )
```

Comment:
The render+commit step here at the end might be good in a separate function, perhaps one that calls the non-sideffecting bits from the first 2/3 of the function?

Reply:
Re-factored.